content/docs/dvclive/api-reference/live/log_param.md

```diff
@@ -0,0 +1,70 @@
+# Live.log_param()
+
+Logs the given parameter `val` associating it with the given `name`.
+
+```py
+ def log_param(name: str, val: ParamLike):
+```
+
+Also see `Live.log_params()`.
+
+## Usage
+
+```py
+from dvclive import Live
+
+live = Live()
+
+live.log_param("lr", 1e-3)
+live.log_param("n_iter", 50)
+```
+
+<admon type="note">
+
+When using `Live(resume=True)`, calling `log_param()` after resuming a session
+will be a no-op.
```

Comment:
Sorry for coming back to this at the docs stage, but what was the motivation for this?

Why having different params after `resume=True` is not considered valid?

cc @dberenbaum

Reply:
I'd find it a bit odd to start one experiment with a set of parameters and resume it with another. In the context of experiment tracking, what set of parameters would be considered valid?

Reply:
I agree having a no-op might be odd. Alternatives include
- overwriting `params.yaml` with the most recent one
- reading the existing `params.yaml` and comparing the parameters to the ones being logged and warning or failing when parameters do not match.
- ...

Reply:
Note that `resume=True` was mainly introduced for `dvc checkpoints`, where one of the motivations is to resume from a checkpoint after modifying some parameters (https://dvc.org/doc/user-guide/experiment-management/checkpoints#starting-from-an-existing-checkpoint).

Outside `dvc checkpoints`, `resume=True` is just something I am not sure we should handle at all 😅, but we do. 

Your concern is valid but I guess we should let the users decide how to handle it. It is only a problem if you execute the training script without persisting the experiment.
If you `resume=True` from a persisted experiment, then the traceability is clear and it makes sense to allow overriding params.

Reply:
> Alternatives include

@dtrifiro what if we change https://github.com/iterative/dvclive/blob/afa21a17c89c048a0e93c0dfbc7c57f8fbed5c83/src/dvclive/live.py#L230 to :

```python
if param_name in self._params and not self._resume:
```

Reply:
This could be ok, although it might allow to log the same parameter twice when using resume.

Reply:
This will only raise when logging the same parameter twice on step 0, although I feel like it's over-complicating this a bit
```python
            if param_name in self._params and not (
                self._resume and self.get_step()
            ):
```

Reply:
> Note that `resume=True` was mainly introduced for `dvc checkpoints`, where one of the motivations is to resume from a checkpoint after modifying some parameters (https://dvc.org/doc/user-guide/experiment-management/checkpoints#starting-from-an-existing-checkpoint).

This is a good point, so if it's easy to accommodate, let's do it. I don't mind generally allowing users to override the dvclive values even if it could lead to dangerous/unexpected results. Users can mostly manage this themselves, and it's probably better to provide flexibility than enforce some strict ideas of correctness.

However, I don't feel the use case is that important, so let's not go crazy addressing this workflow. The idea of changing parameters mid-experiment has always seemed a little contrived to me and not something I have seen users asking for.

Reply:
What would the behavior be if you remove https://github.com/iterative/dvclive/blob/afa21a17c89c048a0e93c0dfbc7c57f8fbed5c83/src/dvclive/live.py#L223-L227? Any parameters that are already logged would fail, but new parameters could still be logged (before or after resume)? I think that's fine if so. Again, I don't think we need to worry too much about either enforcing correctness or addressing every possible workflow for now.

Reply:
Also fine: dropping `ParameterAlreadyLoggedError` altogether and leaving this entirely to users. I probably prefer that most. This kind of error is a little more important in metrics logging where users may forget to increment the step. 

Sorry for the lengthy responses here.

Reply:
>What would the behavior be if you remove [...] Any parameters that are already logged would fail, but new parameters could still be logged (before or after resume)? I think that's fine if so

The idea behind it is to be being able to have an idempotent training script, which you can run to resume training in case files already exist. If we drop that, we'll break this workflow (logging the same parameter twice will error out), unless we decide to drop `ParameterAlreadyLoggedError`, though I fear this might make for some annoying and hard to discover bugs when logging (by mistake) different values to the same key.

Reply:
I think we are overthinking this and should start by dropping `ParameterAlreadyLoggedError`. I think users can handle checking whether they log different values to the same key and would not expect DVCLive to be responsible for checking this.

Reply:
Got rid of it both here and in dvclive: https://github.com/iterative/dvclive/pull/316